manila/share/drivers/netapp/dataontap/client/client_cmode.py

```diff
@@ -4086,18 +4125,61 @@ def prune_deleted_volumes(self):
                     .get_child_by_name('volume-id-attributes')
                     .get_child_content('owning-vserver-name'))
 
+            LOG.debug('Found volume %(vv)s in state  %(vs)s', {
+                      'vv': volume_name, 'vs': volume_state})
             if volume_state == 'offline':
+                clones = self.get_clones_of_parent_volume(vserver, volume_name)
+                if clones:
+                    # Found parent volume which has multiple clone childs, so
+                    # make volume online and start clone splits.
+                    client = copy.deepcopy(self)
+                    client.set_vserver(vserver)
+                    try:
+                        client.volume_online(volume_name)
+                    except Exception:
+                        LOG.error("Volume online failed for "
+                                  "volume %s", volume_name)
+
+                    for clone in clones:
+                        try:
+                            client.volume_online(clone)
+                        except Exception:
+                            LOG.error("Volume online failed for "
```

Comment:
This will likely produce a lot of errors.
Can we ignore the error if the volume was already online? Or make it a debug log in that case at least?

Reply:
the API does not return error code that signify volume is already online. 

Possible reply- Errno

Errno Description
EINTERNALERROR
EVOLUMEPLEXINVALID
EVSERVERNOTFOUND
EVSERVER_OP_NOT_ALLOWED
EOBJECTNOTFOUND
EMULTIPLEVOLSFOUND
EVOLUME_UNSUPPORTED_OPTION_VMALIGN
EAGGR_UNSUPPORTED_OPTION_CONTROLLER_REPLACEMENT_TRANSITION
EVOLUME_UNSUPPORTED_OPTION_SIDL

Reply:
Okay, we will see then and may need to parse the error message ;)